src/gc/gc.cpp

```diff
@@ -308,72 +308,55 @@ void GCStatistics::DisplayAndUpdate()
 
 #endif // GC_STATS
 
-#ifdef BIT64
-#define TOTAL_TIMES_TO_SHIFT 6
-#else
-#define TOTAL_TIMES_TO_SHIFT 5
-#endif // BIT64
-
 size_t round_up_power2 (size_t size)
 {
-    unsigned short shift = 1;
-    size_t shifted = 0;
-
-    size--;
-    for (unsigned short i = 0; i < TOTAL_TIMES_TO_SHIFT; i++)
-    {
-        shifted = size | (size >> shift);
-        if (shifted == size)
-        {
-            break;
-        }
-
-        size = shifted;
-        shift <<= 1;
-    }
-    shifted++;
-
-    return shifted;
+    // Get the 0-based index of the most-significant bit in the size.
+    // If the call failed (because `size` is zero), return 0.
+    DWORD msb_index;
+    if (0 ==
+#ifdef BIT64
+        BitScanReverse64(
+#else
+        BitScanReverse(
+#endif
+            &msb_index, size)) { return 0; }
+
+    // If `size` is already a power-of-two value, return it as-is.
+    // Otherwise, round it up to the next power-of-two by shifting
+    // it's MSB upwards one position.
+    size_t msb_mask = static_cast<size_t>(1) << msb_index;
+    // Always do this so the compiler uses e.g. cmov (x86) or csel (arm) below instead of a branch.
+    size_t rounded_up = msb_mask << 1;
+    return
+        msb_mask == size
+        ? size
+        : rounded_up;
```

Comment:
That's an interesting approach I hadn't considered -- calculating ``msb_index`` for ``size-1`` means ``msb_index`` will be ``31``/``63`` (depending on platform), so left-shifting ``2`` by that will overflow and we'll still return zero. I'll make this change and update the PR.

Reply:
Strictly speaking, 0 should be rounded up to 1 since 0 is not a power of two.  However, the original code returned 0 in this case and I think it is OK to keep that behavior.